llvm/utils/git/code-lint-helper.py

```diff
@@ -124,133 +96,260 @@ def create_comment_text(warning: str, cpp_files: List[str]) -> str:
+            print(f"Git diff failed: {diff_proc.stderr}")
+            return None
+
+        diff_content = diff_proc.stdout
+        if not diff_content.strip():
+            if args.verbose:
+                print("No diff content found")
+            return None
+
+        tidy_diff_cmd = [
+            "clang-tools-extra/clang-tidy/tool/clang-tidy-diff.py",
+            "-path",
+            args.build_path,
+            "-p1",
+            "-quiet",
+        ]
 
-    diff_content = diff_proc.stdout
-    if not diff_content.strip():
-        print("No diff content found")
+        if args.verbose:
+            print(f"Running clang-tidy-diff: {' '.join(tidy_diff_cmd)}")
+
+        proc = subprocess.run(
+            tidy_diff_cmd,
+            input=diff_content,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+            check=False,
+        )
+
+        clean_output = self._clean_clang_tidy_output(proc.stdout.strip())
+        return clean_output
+
+    def _clean_clang_tidy_output(self, output: str) -> Optional[str]:
+        if not output or output == "No relevant changes found.":
+            return None
+
+        lines = output.split("\n")
+        cleaned_lines = []
+
+        for line in lines:
+            if line.startswith("Running clang-tidy in") or line.endswith("generated."):
+                continue
+
+            idx = line.rfind("llvm-project/")
+            if idx != -1:
+                line = line[idx + len("llvm-project/") :]
+
+            cleaned_lines.append(line)
+
+        if cleaned_lines:
+            return "\n".join(cleaned_lines)
         return None
 
-    tidy_diff_cmd = [
-        "clang-tools-extra/clang-tidy/tool/clang-tidy-diff.py",
-        "-path",
-        args.build_path,
-        "-p1",
-        "-quiet",
-    ]
-
-    if args.verbose:
-        print(f"Running clang-tidy-diff: {' '.join(tidy_diff_cmd)}")
-
-    proc = subprocess.run(
-        tidy_diff_cmd,
-        input=diff_content,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        text=True,
-        check=False,
-    )
 
-    return clean_clang_tidy_output(proc.stdout.strip())
+class Doc8LintHelper(LintHelper):
+    name = "doc8"
+    friendly_name = "documentation linter"
 
+    def instructions(self, doc_files: List[str], args: LintArgs) -> str:
+        files_str = " ".join(doc_files)
+        return f"doc8 -q {files_str}"
 
-def run_linter(changed_files: List[str], args: LintArgs) -> tuple[bool, Optional[dict]]:
-    changed_files = [arg for arg in changed_files if "third-party" not in arg]
+    def filter_changed_files(self, changed_files: List[str]) -> List[str]:
+        filtered_files = []
+        for filepath in changed_files:
+            _, ext = os.path.splitext(filepath)
+            if ext not in (".rst"):
+                continue
+            if not filepath.startswith("clang-tools-extra/docs/clang-tidy/checks/"):
```

Comment:
Are there any obstacles in linting all docs?
```suggestion
            if not filepath.startswith("clang-tools-extra/docs/clang-tidy/"):
```